pennylane/ops/qubit/hamiltonian.py

```diff
@@ -625,11 +625,16 @@ def compare(self, other):
         >>> ob1.compare(ob2)
         False
         """
-        if isinstance(other, (Hamiltonian, qml.ops.LinearCombination)):
+        if isinstance(other, Hamiltonian):
             self.simplify()
             other.simplify()
             return self._obs_data() == other._obs_data()  # pylint: disable=protected-access
 
+        if isinstance(other, qml.ops.LinearCombination):
+            self.simplify()
+            other = other.simplify()
+            return self._obs_data() == other._obs_data()  # pylint: disable=protected-access
```

Comment:
I think you can remove this with the latest updates, old Hamiltonians and LinearCombinations shouldnt "see" each other in any of those tests